src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -112,10 +112,6 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
     if (isTerminalState(status)) {
       liveState.removeRunningTask(status.getTaskId());
       persistentState.removeTaskId(status.getTaskId().getValue());
-      if (reconciliationComplete()) {
-        correctCurrentPhase();
-      }
-
```

Comment:
+1 on putting this back in. Definitely necessary for dropping down to a JN/NN phase when one of those dies, or to advance out of RECONCILING if the first status update after the timeout expires is a terminal (not necessarily from a reconcilation update). There should be a unit test or two that breaks if this is removed.